File: java/55.jump-game.java
class Solution {
  public boolean canJump(int[] nums) {
      int maxLength = 0;
      for (int i = 0; i < nums.length; i++) {
          if (i > maxLength) {
              break;
          }
          maxLength = Math.max(maxLength, i + nums[i]);
      }
      if (maxLength >= nums.length - 1) {
          return true;
      }
      return false;
  }
}

File: python3/55.jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        if not nums:
            return False
        maxReach = 0
        for index in range(len(nums)):
            if index <= maxReach:
                maxReach = max(maxReach, index + nums[index])
        return maxReach >= len(nums) - 1